src/scaler/spreadscaler.rs

```diff
@@ -0,0 +1,984 @@
+                host_id: host_id_one.to_string()
+            }))
+            .await?
+            .is_empty());
+        assert!(blobby_spreadscaler
+            .handle_event(Event::ProviderStopped(ProviderStopped {
+                contract_id: "".to_string(),
+                instance_id: "".to_string(),
+                link_name: "".to_string(),
+                public_key: "".to_string(),
+                reason: "".to_string(),
+                host_id: host_id_two.to_string()
+            }))
+            .await?
+            .is_empty());
+        assert!(blobby_spreadscaler
+            .handle_event(Event::LinkdefSet(LinkdefSet {
+                linkdef: Linkdef::default()
+            }))
+            .await?
+            .is_empty());
+        assert!(blobby_spreadscaler
+            .handle_event(Event::LinkdefDeleted(LinkdefDeleted {
+                linkdef: Linkdef::default()
+            }))
+            .await?
+            .is_empty());
+
+        let cmds = blobby_spreadscaler.reconcile().await?;
+        assert_eq!(cmds.len(), 2);
+
+        for cmd in cmds.iter() {
+            match cmd {
+                Command::StartActor(start) => {
+                    assert_eq!(start.host_id, host_id_three.to_string());
+                    assert_eq!(start.count, 3);
+                    assert_eq!(start.reference, blobby_ref);
+                }
+                Command::StopActor(stop) => {
+                    assert_eq!(stop.host_id, host_id_two.to_string());
+                    assert_eq!(stop.count, 16);
+                    assert_eq!(stop.actor_id, blobby_id);
+                }
+                _ => panic!("Unexpected command in spreadscaler list"),
+            }
+        }
+
+        // Stop an instance of an actor, and expect a modified list of commands
+
+        let modifying_event = ActorStopped {
+            instance_id: "IAMANINSTANCE".to_string(),
+            public_key: blobby_id.to_string(),
+            host_id: host_id_two.to_string(),
+        };
+
+        worker
+            .do_work(ScopedMessage::<Event> {
+                lattice_id: lattice_id.to_string(),
+                inner: Event::ActorStopped(modifying_event.clone()),
+                acker: None,
+            })
+            .await
+            .expect("should be able to handle an event");
+
+        let cmds = blobby_spreadscaler
+            .handle_event(Event::ActorStopped(modifying_event))
+            .await?;
+        assert_eq!(cmds.len(), 2);
+
+        for cmd in cmds.iter() {
+            match cmd {
+                Command::StartActor(start) => {
+                    assert_eq!(start.host_id, host_id_three.to_string());
+                    assert_eq!(start.count, 3);
+                    assert_eq!(start.reference, blobby_ref);
+                }
+                Command::StopActor(stop) => {
+                    assert_eq!(stop.host_id, host_id_two.to_string());
+                    assert_eq!(stop.count, 15);
```

Comment:
This assertion currently fails since the `ActorStopped` event doesn't modify the host state, which is what the spreadscaler reads from. Considering either refactoring the scaler (not desired) or adding host state modifications to the event worker (desired)

Reply:
edit: added modifications to event worker in https://github.com/wasmCloud/wadm/pull/75/commits/01a6194d0f954f825cae31fb34a021ce1b2604eb